src/go/rpk/pkg/cli/topic/analyze.go

```diff
@@ -0,0 +1,743 @@
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+
+	sections.Add(secBatchSize, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchSizes {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+	return nil
+}
+
+type kvTPInfo struct {
+	TopicPartition topicPartition      `json:"topicPartition" yaml:"topic_partition"`
+	Info           *topicPartitionInfo `json:"info" yaml:"info"`
+}
+
+type rawResults struct {
+	TopicPartitionInfo []kvTPInfo `json:"topicPartitionInfo" yaml:"topic_partition_info"`
+	TimeRange          timeRange  `json:"timeRange" yaml:"time_range"`
+}
+
+func (cr collectResult) printRawResults(f config.OutFormatter, w io.Writer) (bool, error) {
+	// Json doesn't seem to like serializing the map types. So convert it to a slice here.
+	kvs := []kvTPInfo{}
+	for k, v := range cr.TopicPartitionInfo {
+		kvs = append(kvs, kvTPInfo{k, v})
+	}
+
+	res := rawResults{kvs, cr.TimeRange}
+	if isText, _, t, err := f.Format(res); !isText {
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Fprintln(w, t)
+		return true, nil
+	}
+	return false, nil
+}
+
+const helpAnalyze = `Analyze topics.
+
+This command consumes records from the specified topics in order to determine
+characteristics of the topic like batch rate and batch size.
+
+Using the --format flag with either JSON or YAML prints all of the metadata collected.
+
+TOPICS
+
+Topics can either be listed individually or by regex via the --regex flag (-r).
+
+For example,
+
+    analyze foo bar            # describe topics foo and bar
+    analyze -r '^f.*' '.*r$'   # describe any topic starting with f and any topics ending in r
+    analyze -r '*'             # describe all topics
+    analyze -r .               # describe any one-character topics
+
+TIME RANGE
+
+The --time-range flag allows for specifying what time range to consume from.
+The time range should be of the following format:
+
+    t1:t2    consume from timestamp t1 until timestamp t2
+
+There are a few options for timestamps, with each option being evaluated
+until one succeeds:
+
+    13 digits             parsed as a unix millisecond
+    9 digits              parsed as a unix second
+    YYYY-MM-DD            parsed as a day, UTC
+    YYYY-MM-DDTHH:MM:SSZ  parsed as RFC3339, UTC; fractional seconds optional (.MMM)
+    end                   for t2 in @t1:t2, the current end of the partition
+    -dur                  a negative duration from now or from a timestamp
+    dur                   a positive duration from now or from a timestamp
+
+Durations can be relative to the current time or relative to a timestamp.
+If a duration is used for t1, that duration is relative to now.
+If a duration is used for t2, if t1 is a timestamp, then t2 is relative to t1.
```

Comment:
```suggestion
If a duration is used for t2, and t1 is a timestamp, then t2 is relative to t1.
```

Reply:
Will change.

Note that a lot of the wording I used in the description is close to or taken from `rpk topic describe` and `rpk topic consume`. You may want to open up a PR to make similar changes.